packages/entities/entities-shared/src/composables/useDebouncedFilter.ts

```diff
@@ -93,20 +98,58 @@ export default function useDebouncedFilter(
         error.value = ''
         validationError.value = ''
 
-        // If user has typed info in the query field
-        let currUrl = url + '' // clone
-        if (query) {
-          currUrl += `/${query}`
-        }
+        if (config.app === 'konnect') { // KoKo only supports exact match
+          // If user has typed info in the query field
+          let currUrl = url + '' // clone
+          if (query) {
+            currUrl += `/${query}`
+          }
 
-        const { data }: Record<string, any> = await axiosInstance.get(`${currUrl}?size=${size}`)
+          const { data }: Record<string, any> = await axiosInstance.get(`${currUrl}?size=${size}`)
 
-        if (keys.fetchedItemsKey in data) {
-          results.value = data[keys.fetchedItemsKey]
-        } else if (data?.id) { // exact match
-          results.value = [data]
-        } else {
-          results.value = []
+          if (keys.fetchedItemsKey in data) {
+            results.value = data[keys.fetchedItemsKey]
+          } else if (data?.id) { // exact match
+            results.value = [data]
+          } else {
+            results.value = []
+          }
+        } else { // Admin API supports filtering on specific fields
+          const promises = []
+
+          if (isValidUuid(query || '') && keys.searchKeys.includes('id')) {
+            // If query is a valid UUID, do the exact search
+            promises.push((async () => {
+              const res: Record<string, any> = await axiosInstance.get(`${url}/${query}`)
+              return [res[keys.fetchedItemsKey]]
+            })())
+          } else if (query) {
+            // Search on fields with backend filtering support
+            promises.push(
+              ...keys.searchKeys
+                .filter(key => key !== 'id')
+                .map(async key => {
+                  const { data } = await axiosInstance.get(`${url}?${key}=${query}`)
+                  return data[keys.fetchedItemsKey]
+                }),
+            )
+          }
+
+          if (query) {
```

Comment:
Maybe we can hoist this `if` to line 118 to prevent issuing unnecessary network requests

Reply:
If `query` is empty, the `promises` array will be empty as well. But hoisting this `if` up does make the code cleaner, so I did it.